### PR TITLE
Split "compiler" from "caller"

### PR DESCRIPTION
**Pull request by gitgabrio**
Scope of this PR is to split the actual compiler from the invoker. This would allow easy reusability of the "compiler" class while keeping the "invoker" signature the same. 
The overall idea is to allow different "client" code (for ex. Maven plugin) to invoke the compiler; "client" code should then only provide the "configuration".
Components would be:

1. Compiler (ListeningCompiler)
2. Configuration (Gwt3Options implementations)
3. Invoker/configurator ()DevMode, maven plugin other).

All the "compiler" code has been moved to the ListeningCompiler class, while the DevMode one has been kept as "configurer" to allow direct command line invocation.
For the same scope, Gwt3Options has been split in interface and actual implementation.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-  <groupId>com.vertispan.j2cl</groupId>
+  <groupId>net.cardosi.j2cl</groupId>
```
Comment (niloc132) [inline pom.xml]: Hmm, maybe don't merge that back ;)

File: pom.xml [hunk 2]
```diff
@@ -121,6 +121,7 @@
       <url>https://repo.vertispan.com/j2cl</url>
     </snapshotRepository>
   </distributionManagement>
+
```
Comment (niloc132) [inline pom.xml]: ?

**Comment by niloc132** [comment]
I didnt actually diff the changes between DevMode and the new class, but will do so before merging. I'll hold off on more changes (found a few bugs...) until this lands.

File: pom.xml (hunk 2 above)
Comment (gitgabrio) [reply pom.xml]: Sorry? Do not understand your question

File: pom.xml (hunk 1 above)
Comment (gitgabrio) [reply pom.xml]: Ok.

**Comment by gitgabrio** [comment]
@niloc132 
Pushed correct groupid

**Pull request merged**